storage/innobase/dict/dict0dict.cc

```diff
@@ -1564,7 +1564,8 @@ dict_table_rename_in_cache(
 					strcat(foreign->id,
 					       old_id + ut_strlen(old_name));
 				} else {
-					sprintf(strchr(foreign->id, '/') + 1,
+					snprintf(strchr(foreign->id, '/') + 1,
+						sizeof(strchr(foreign->id, '/') + 1),
```

Comment:
dict_foreign_t.id is ptr, not a char array.

Reply:
Reverted back to sprintf function